cranelift/codegen/src/legalizer/mod.rs

```diff
@@ -285,17 +284,17 @@ fn expand_cond_trap(
 
     // Replace trap instruction by the inverted condition.
     if trapz {
-        func.dfg.replace(inst).brnz(arg, new_block_resume, &[]);
+        func.dfg
+            .replace(inst)
+            .brif(arg, new_block_resume, &[], new_block_trap, &[]);
     } else {
-        func.dfg.replace(inst).brz(arg, new_block_resume, &[]);
+        func.dfg
+            .replace(inst)
+            .brif(arg, new_block_trap, &[], new_block_resume, &[]);
     }
 
-    // Add jump instruction after the inverted branch.
-    let mut pos = FuncCursor::new(func).after_inst(inst);
-    pos.use_srcloc(inst);
```

Comment:
I don't know exactly what `use_srcloc` does but does using `.replace(inst)` above effectively do the same thing as this?

Reply:
I think that that the effects of `use_srcloc` are lost in my new version, as any instructions inserted using `pos`will not inherit that location. Thanks for catching this!